### PR TITLE
[service.library.data.provider] v0.4.0

### PR DESCRIPTION
**Pull request by mikesilvo164**
### Description
Python 3 compatible update for service.library.data.provider

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by mikesilvo164** [comment]
Not sure if this should go to Leia and up since the old version currently works fine

**Comment by basrieter** [comment]
@mikesilvo164 what's the status on this PR? Did you update your code to make Travis CI happy?

**Comment by mikesilvo164** [comment]
Sorry took so long was on a break from all things Kodi. Everything looks good to me just need to wait for script.module.routing to be available in current repository.

File: service.library.data.provider/.gitignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+*.pyc
```
Comment (basrieter) [inline service.library.data.provider/.gitignore]: Please remove the .gitignore file from the PR

File: service.library.data.provider/changelog.txt [hunk 2]
```diff
@@ -0,0 +1,44 @@
+[B]0.1.6[/B]
```
Comment (basrieter) [inline service.library.data.provider/changelog.txt]: Not needed in the PR. Please remove. Use the `<news>` tag in addon.xml

File: service.library.data.provider/resources/lib/library.py [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#    This script is based on service.skin.widgets
+#    Thanks to the original authors
+
+import sys
```
Comment (basrieter) [inline service.library.data.provider/resources/lib/library.py]: the `import sys` is not used.

**Comment by basrieter** [comment]
This also requires `script.module.routing` which is not yet in Matrix.

**Comment by mikesilvo164** [comment]
Requested changes pushed 😃

**Comment by basrieter** [comment]
Just waiting for the mirrors to include https://github.com/xbmc/repo-scripts/pull/1349

**Pull request merged**